pyaedt/generic/ibis_reader.py

```diff
@@ -0,0 +1,748 @@
+            Model selector item.
+
+        """
+
+        item = ModelSelectorItem()
+        i_start = current_line.index(" ", 1)
+
+        if i_start > 0:
+            item.name = current_line[i_start:].strip()
+            item.description = current_line[i_start:].strip()
+
+        return item
+
+    # Component
+    def read_component(self, ibis, current_line, file):
+        """Extracts component's info.
+
+        Parameters
+        ----------
+        ibis : :class:`pyaedt.generic.ibis_reader.Ibis`
+            ibis object containing all info.
+        current_line : str
+            Current line content.
+        file : TextIO
+            File's stream.
+
+        """
+
+        if self.IsStartedWith(current_line, "[Component] ") != True:
+            return
+
+        component = Component()
+        component.name = self.get_component_name(current_line)
+        current_line = file.readline()
+
+        if self.IsStartedWith(current_line, "[Manufacturer]") == True:
+            component.manufacturer = current_line.replace("[Manufacturer]", "").strip()
+
+        while True:
+            current_line = file.readline()
+            if self.IsStartedWith(current_line, "[Package]") == True:
+                break
+
+        self.fill_package_info(component, current_line, file)
+
+        # [pin]
+        while self.IsStartedWith(current_line, "[Pin] ") != True:
+            current_line = file.readline()
+
+        while True:
+            current_line = file.readline()
+            if self.IsStartedWith(current_line, "|") == True:
+                break
+
+        current_line = file.readline()
+
+        while self.IsStartedWith(current_line, "|") == False:
+            pin = self.make_pin_object(current_line, component.name, ibis)
+            component.pins[pin.name] = pin
+            current_line = file.readline()
+            if current_line == "":
+                break
+
+        ibis.components[component.name] = component
+
+    def fill_package_info(self, component, current_line, file):
+        """Extracts model's info.
+
+        Parameters
+        ----------
+        component : :class:`pyaedt.generic.ibis_reader.Component`
+            Current line content.
+        current_line : str
+            Current line content.
+        file : TextIO
+            File's stream.
+
+        """
+        while self.IsStartedWith(current_line, "|") == True or self.IsStartedWith(current_line, "[") == True:
+            current_line = file.readline()
+
+        if self.IsStartedWith(current_line, "R_pkg") == True:
+            component.R_pkg = current_line.strip()
+            current_line = file.readline()
+        elif self.IsStartedWith(current_line, "L_pkg") == True:
+            component.L_pkg = current_line.strip()
+            current_line = file.readline()
+        elif self.IsStartedWith(current_line, "C_pkg") == True:
+            component.C_pkg = current_line.strip()
+
+    def get_component_name(self, line):
```

Comment:
This method is a candidate for `@classmethod` or to be moved out of the class as a standard function.

Reply:
It is now a ``@classmethod``.